src/aws_s3_presigned_url.erl

```diff
@@ -23,23 +23,22 @@ make_presigned_v4_url(Client0, Method, ExpireSeconds, Bucket, Key) ->
     Client = Client0#{service => <<"s3">>},
     SecurityToken = maps:get(token, Client, undefined),
     AccessKeyID = maps:get(access_key_id, Client),
+    SecretAccessKey = maps:get(secret_access_key, Client),
     Region = maps:get(region, Client),
+    Service = maps:get(service, Client),
     Host = build_host(<<"s3">>, Client, Bucket),
-    {BaseHost, URL0} = build_url(Host, Path, Client, Bucket),
+    URL = build_url(Host, Path, Client, Bucket),
     Now = calendar:universal_time(),
-    ShortDate = list_to_binary(ec_date:format("Ymd", Now)),
-    Credential = make_credential(AccessKeyID, ShortDate, Region),
-    Headers0 = [{<<"Host">>, Host},
-                {<<"X-Amz-Algorithm">>, <<"AWS4-HMAC-SHA256">>},
-                {<<"X-Amz-Credential">>, Credential},
-                {<<"X-Amz-Date">>, list_to_binary(ec_date:format("YmdTHisZ", Now))},
-                {<<"X-Amz-Expires">>, integer_to_binary(ExpireSeconds)},
-                {<<"X-Amz-SignedHeaders">>, <<"host">>}
+    Options0 = [ {ttl, ExpireSeconds}
+               , {body_digest, <<"UNSIGNED-PAYLOAD">>}
                ],
-    Headers = maybe_add_security_token(Headers0, SecurityToken),
-    Signature = sign_request(Client, MethodBin, BaseHost, aws_request:add_query(URL0, Headers), [{<<"Host">>, BaseHost}], Now, <<"UNSIGNED-PAYLOAD">>),
-    FinalHeaders = lists:keysort(1, [{<<"X-Amz-Signature">>, Signature} | Headers]),
-    {ok, aws_request:add_query(URL0, FinalHeaders)}.
+    Options = case SecurityToken of
+                undefined ->
+                  Options0;
+                _ ->
+                  [{session_token, hackney_url:urlencode(SecurityToken)} | Options0]
```

Comment:
The issue here seems to be that the CanonicalRequest that `aws_signature` builds up does not urlencode the `session_token` while AWS itself does when it calculates it's own CanonicalRequest in order to compare signatures. Rather than making the change in `aws_signature` and potentially breaking `aws-elixir` I've moved the change in here. 

@philss Since you're closest to `aws-elixir` to my knowledge, are you aware of any issues regarding this? Nor do you seem to urlencode the token prior to setting the header (See: [signature.ex#L16](https://github.com/aws-beam/aws-elixir/blob/cadf6502f238990f95e0e51190ecb08ce5e4a0a8/lib/aws/signature.ex#L16)) which confuses me a bit. Perhaps this is only necessary for presigned urls and hence [my other comment](https://github.com/aws-beam/aws-erlang/pull/126/files#r1107781493) should be reverted out?

Reply:
Shouldn't we try to use https://github.com/aws-beam/aws_signature/blob/main/src/aws_signature.erl#L101 here?

Reply:
@onno-vos-dev I'm not sure, but I think it may be because this is more relevant for the URLs.

I think @jadeallenx 's suggestion makes sense! That function accepts a `session_token` as option, but it does not encode it, for what I could see. Maybe we didn't test that part.

Reply:
> Shouldn't we try to use https://github.com/aws-beam/aws_signature/blob/main/src/aws_signature.erl#L101 here?

That is the one I'm using? :-) I'm passing in `options` with the `session_token` but apparently it requires urlencoding to pass. And that feels kinda odd to me but since I don't have an Elixir example at hand I'm not sure if this is "correct" or not :-) It seems to be.

Reply:
OK, sorry - I missed that you were already calling that function